lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1468,10 +1468,18 @@ def customize_customvalues(self, vm_obj, config_spec):
                 self.module.exit_json(msg="customvalues items required both 'key' and 'value fields.")
 
             # If kv is not kv fetched from facts, change it
-            if kv['key'] not in facts['customvalues'] or facts['customvalues'][kv['key']] != kv['value']:
+            valuetype = type(kv['value'])
+            if valuetype is bool or valuetype is int:
+                specifiedvalue = str(kv['value']).upper()
+                comparisonvalue = facts['customvalues'][kv['key']].upper()
```

Comment:
Shouldn't you check first if the `kv['key']` is in `facts['customvalues']`? (or use `get`, like you do below)

Reply:
Fixed.